app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -129,25 +133,65 @@ if [ x$ID = xrhel ] || [ x$ID = xcentos ] || [ x$ID = xol ]; then
       yum install -y --setopt=obsoletes=0 subscription-manager
     fi
 
-    CONSUMER_RPM=$(mktemp --suffix .rpm)
-    curl --silent --show-error --output $CONSUMER_RPM <%= subscription_manager_configuration_url(hostname: @url_host) %>
+    # Prepare SSL certificate
+    cp -f $SSL_CA_CERT $KATELLO_SERVER_CA_CERT
+    chmod 644 $KATELLO_SERVER_CA_CERT
+
+    # Prepare subscription-manager
+    yum remove -y katello-ca-consumer*
 
-    # Workaround for systems with enabled FIPS,
-    # where installation of RPM generated on RHEL7 cause 'no digest' error
-    # See https://projects.theforeman.org/issues/32068
-    if [ "$(cat /proc/sys/crypto/fips_enabled)" = "1" ]; then
-        rpm -ivh --nodigest --nofiledigest $CONSUMER_RPM
+    if ! [ -x "$(command -v subscription-manager)" ] ; then
+      if [ "${VERSION_ID%.*}" -gt 7 ]; then
+        dnf install -y subscription-manager
```

Comment:
I might have said that before, but it feels overly complicated to have this if/else for version > 7 here, but not using the same in L141 where you remove ca-consumer, or L132/133 where you poke around sub-man.

given `yum` is an alias for `dnf` on EL8, I'd probably just use `yum` everywhere for now, and later introduce a central function that does the if/else?

Reply:
> I'd probably just use yum everywhere for now, and later introduce a central function that does the if/else?

It was like that before but it wasn't accepted well by community so I had to change it to the current logic

Reply:
I had suggested trying to future proof this more. I do get the point @evgeni is raising here, this logic should be applied in all cases of the use of yum or dnf and thus having a function to wrap the logic so it can be re-used everywhere is much cleaner.  Whether thats here now or in a follow up.

If we can get an issue that tracks making that update to determine yum vs. dnf using a single function I'm good with just using yum for now.

Neither option blocks for me. But the lack of consistency ought to be tracked and addressed.

Reply:
Yeah, I'm all for future proofing this. But then *all* calls the $packagemanager need to be.